native/swift/Tests/wordpress-api/LoginTests.swift

```diff
@@ -7,37 +7,280 @@ import Testing
+    ])
+    func testAdminUrlProvided(_ provided: String, _ expected: String) async throws {
+        let parsedUrl = try await client.findLoginUrl(forSite: provided)
+        #expect(expected == parsedUrl.url())
+    }
+
+    @Test("Login Spec Example 4: HTTP URL with HTTPS Support")
+    func testAutoHttpsSupport() async throws {
+        let parsedUrl = try await client.findLoginUrl(forSite: "http://vanilla.wpmt.co")
+        #expect("https://vanilla.wpmt.co/wp-admin/authorize-application.php" == parsedUrl.url())
+    }
+
+    @Test("Login Spec Example 5: HTTP-only site")
+    func testHttpOnlySite() async {
+        await #expect(performing: {
+            _ = try await client.findLoginUrl(forSite: "http://no-https.wpmt.co")
+        }, throws: { error in
+            let reason = try #require(try self.getApplicationPasswordsNotSupportedReason(from: error))
+
+            guard case .applicationPasswordsDisabledForHttpSite = reason else {
+                Issue.record("The reason should be .ApplicationPasswordsDisabledForHttpSite")
+                return false
+            }
+
+            return true
+        })
+    }
+
+    @Test("Login Spec Example 6: HTTP-Only Site with Application Password Override")
+    func testHttpOnlySiteWithApplicationPasswordsEnabled() async throws {
+        let parsedUrl = try await client.findLoginUrl(forSite: "http://no-https-with-application-passwords.wpmt.co")
+        #expect("http://no-https-with-application-passwords.wpmt.co/wp-admin/authorize-application.php" == parsedUrl.url())
+    }
+
+    @Test("Login Spec Example 7: CDN-Cached Site")
+    func testAggressivelyCachedSiteWithNoLinkheader() async throws {
+        let parsedUrl = try await client.findLoginUrl(forSite: "https://aggressive-caching.wpmt.co")
+        #expect("https://aggressive-caching.wpmt.co/wp-admin/authorize-application.php" == parsedUrl.url())
+    }
+
+    @Test("Login Spec Example 8: Site with Application Passwords Disabled by WordFence")
+    func testSiteWithApplicationPasswordsDisabledByWordFence() async throws {
+        await #expect(performing: {
+            _ = try await client.findLoginUrl(forSite: "https://wordfence.wpmt.co")
+        }, throws: { error in
+            let reason = try #require(try self.getApplicationPasswordsNotSupportedReason(from: error))
+
+            guard case .applicationPasswordBlockedByPlugin(plugin: let plugin) = reason else {
+                Issue.record("The reason should be .ApplicationPasswordsDisabledForHttpSite")
+                return false
+            }
+
+            #expect(plugin.name == "Wordfence")
+
+            return true
+        })
+    }
+
+    @Test("Login Spec Example 9: Not a WordPress Site")
     func testNotWordPressSite() async throws {
-        let client = WordPressLoginClient(urlSession: .shared)
+        await #expect(performing: {
+            _ = try await client.findLoginUrl(forSite: "https://google.com")
```

Comment:
When the input is "google.com", the returned error is `ParseUrlError.RelativeUrlWithoutBase`. IMO that's an incorrect error. I'd expect the error to be `probablyNotAWordPressSite`.

Reply:
I tested this in Rust with the new `combinedResult` function from #652 and it returned `ProbablyNotAWordPressSite` as expected. Once this PR is updated, the issue should be resolved for Swift as well.

```
// Full error message for `google.com`
FindApiRoot {
    parsed_site_url: ParsedUrl {
        inner: Url {
            scheme: "https",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: Some(
                Domain(
                    "google.com",
                ),
            ),
            port: None,
            path: "/",
            query: None,
            fragment: None,
        },
    },
    find_api_root_failure: ProbablyNotAWordPressSite,
}
```

Reply:
@jkmassel Do you mind adding `google.com` as a test case, which should be fairly easy with `@Test(arguments: ["google.com", "https://google.com"])?